src/main/surfrdUtilsMod.F90

```diff
@@ -190,110 +190,112 @@ subroutine collapse_all_pfts(wt_lunit, wt_nat_patch, natpft_size, wt_cft, cft_si
+                wt_lunit(g,istsoil) = 0._r8
+                wt_nat_patch(g,:) = 0._r8
+                wt_nat_patch(g,noveg) = 1._r8  ! to pass error check below
+             else
+                do n = 1, n_dom_soil_patches
+                   m = max_indices(n)
+                   if (m <= natpft_ub) then
+                      wt_nat_patch(g,m) = wt_nat_patch(g,m) / wt_dom_nat_sum
+                   end if
+                end do
+             end if
+             ! Normalize dominant cft weights to 1; however,
+             ! if non-existent, then merge the nat and crop landunit weights
+             ! and set the cft weights and cft landunit weight to 0.
+             ! Don't bother merging if they're not both > 0.
+             if (wt_dom_cft_sum <= 0._r8) then
+                if (wt_lunit(g,istcrop) > 0._r8 .and. wt_lunit(g,istsoil) > 0._r8) then
+                   wt_lunit(g,istsoil) = wt_lunit(g,istsoil) + wt_lunit(g,istcrop)
                 end if
-                ! Set to .F. regardless of whether dominant pfts and cfts were
-                ! found
-                pftcon%is_pft_known_to_model(m) = .false.
+                wt_lunit(g,istcrop) = 0._r8
+                wt_cft(g,:) = 0._r8
+                wt_cft(g,nc3crop) = 1._r8  ! to pass error check below
+             else
+                do n = 1, n_dom_soil_patches
+                   m = max_indices(n)
+                   if (m >= cft_lb) then
+                      wt_cft(g,m) = wt_cft(g,m) / wt_dom_cft_sum
+                   end if
+                end do
              end if
-          end do
-
-       end do
+             ! Set non-dominant pft and cft weights to 0 for cases where dominant
+             ! pfts or cfts were found
+             do m = 0, maxveg
+                if (.not. any(max_indices == m)) then
+                   if (m <= natpft_ub .and. wt_dom_nat_sum > 0._r8) then
+                      wt_nat_patch(g,m) = 0._r8
+                   else if (m >= cft_lb .and. wt_dom_cft_sum > 0._r8) then
+                      wt_cft(g,m) = 0._r8
+                   end if
+                   ! Set to .F. regardless of whether dominant pfts and cfts were
+                   ! found
+                   pftcon%is_pft_known_to_model(m) = .false.
+                end if
+             end do
 
-       write(iulog,*) 'is_pft_known...', pftcon%is_pft_known_to_model  ! slevis diag
+          end do
 
-       ! Error checks
-       call check_sums_equal_1(wt_lunit, begg, 'wt_lunit', subname)
-       call check_sums_equal_1(wt_cft, begg, 'wt_cft', subname)
-       call check_sums_equal_1(wt_nat_patch, begg, 'wt_nat_patch', subname)
+          ! Error checks
+          call check_sums_equal_1(wt_lunit, begg, 'wt_lunit', subname)
+          call check_sums_equal_1(wt_cft, begg, 'wt_cft', subname)
+          call check_sums_equal_1(wt_nat_patch, begg, 'wt_nat_patch', subname)
 
-       deallocate(max_indices)
+          deallocate(max_indices)
+       end if
```

Comment:
Diffs due to indentation from adding 
`if (cft_size > 0) then  ! The opposite applies only if use_fates`
end here!